de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java

```diff
@@ -145,11 +148,25 @@ protected void save() {
 	 */
 	protected SWTBotTreeItem openEditor(SWTBotTreeItem item) {
 		SWTBotTreeItem newItem = item.doubleClick();
+		waitForEditor(item);
 		bot.saveAllEditors();
 		waitForEditingDomainAndUiThread();
 		return newItem;
 		
 	}
+	
+	/**
+	 * Waits for the editor of the passed item to open.
+	 * The editor is identified using the name of the item,
+	 * so it cant distinguish between editors of items with the same name!
+	 * @param item the item whose editor we are waiting for
+	 */
+	protected void waitForEditor(SWTBotTreeItem item) {
+		String label = item.getText();
+		Matcher<IEditorReference> matcher = withPartName(StringStartsWith.startsWith(label + " -> "));
```

Comment:
Does it make sense to create the editor name here by hand. Shouldn't it be derived from a method actually creating the name in the editor. SO in case the pattern changes, the test case won't fail?